Common/Tests/Utilities/VisualStudioPath.cs

```diff
@@ -16,25 +16,34 @@
 
 using System;
 using System.IO;
+using System.Runtime.InteropServices;
 using Microsoft.VisualStudio.Setup.Configuration;
 
 namespace TestUtilities {
     public static class VisualStudioPath {
         private static Lazy<string> RootLazy { get; } = new Lazy<string>(GetVsRoot);
-        private static Lazy<string> CommonExtensionsLazy { get; } = new Lazy<string>(() => Path.Combine(Root, @"CommonExtensions\"));
-        private static Lazy<string> PrivateAssembliesLazy { get; } = new Lazy<string>(() => Path.Combine(Root, @"PrivateAssemblies\"));
-        private static Lazy<string> PublicAssembliesLazy { get; } = new Lazy<string>(() => Path.Combine(Root, @"PublicAssemblies\"));
+        private static Lazy<string> CommonExtensionsLazy { get; } = new Lazy<string>(() => Root == null ? null : Path.Combine(Root, @"CommonExtensions\"));
+        private static Lazy<string> PrivateAssembliesLazy { get; } = new Lazy<string>(() => Root == null ? null : Path.Combine(Root, @"PrivateAssemblies\"));
+        private static Lazy<string> PublicAssembliesLazy { get; } = new Lazy<string>(() => Root == null ? null : Path.Combine(Root, @"PublicAssemblies\"));
 
         public static string Root => RootLazy.Value;
         public static string CommonExtensions => CommonExtensionsLazy.Value;
         public static string PrivateAssemblies => PrivateAssembliesLazy.Value;
         public static string PublicAssemblies => PublicAssembliesLazy.Value;
 
         private static string GetVsRoot() {
-            var configuration = (ISetupConfiguration2)new SetupConfiguration();
-            var current = (ISetupInstance2)configuration.GetInstanceForCurrentProcess();
-            var path = current.ResolvePath(current.GetProductPath());
-            return Path.GetDirectoryName(path);
+            try {
+                var configuration = (ISetupConfiguration2)new SetupConfiguration();
+                var current = (ISetupInstance2)configuration.GetInstanceForCurrentProcess();
+                var path = current.ResolvePath(current.GetProductPath());
+                return Path.GetDirectoryName(path);
+            } catch (COMException) {
+                var path = Environment.GetEnvironmentVariable($"VisualStudio_IDE_{AssemblyVersionInfo.VSVersion}");
```

Comment:
Is it "class not registered" COM exception for `new SetupConfiguration()`?

Reply:
Yeah. I suspect your work to re-split up the tests will help more, but this change is good enough to get our pass rate back up to a reasonable value.

Reply:
There is one more thing that we can fallback (before or after looking for the variable):

https://github.com/dotnet/roslyn/blob/a3f904c792fbe6b22fad1254b299b8c178340247/src/Workspaces/MSBuildTest/Interop.cs

Reply:
I think we're still not going to find the instance from the current process anyway, so falling back to the environment variable is the best option for our test runs.

We probably need to move the non-UI tests that depend on VS binaries to run in-proc. As I say, I'm sure you'll get this sorted out just fine in the bigger change :)